comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -136,7 +143,34 @@ func (l *listenerCandidate) try() (listeners.ServiceListener, error) {
 
 // newAutoConfig creates an AutoConfig instance and starts it.
 func newAutoConfig(deps dependencies) autodiscovery.Component {
-	ac := createNewAutoConfig(scheduler.NewController(), deps.Secrets, deps.WMeta, deps.TaggerComp, deps.Log, deps.Telemetry)
+	schController := scheduler.CreateNewController()
+	// Non-blocking start of the scheduler controller
+	go func() {
+		instance, found := deps.WMeta.Get()
+		if found {
+			ticker := time.NewTicker(WmetaCheckInterval)
+			defer ticker.Stop()
+			timeout := time.After(MaxWmetaWaitTime)
+			for {
+				select {
+				case <-ticker.C:
+					if instance.IsInitialized() {
```

Comment:
the code looks good, 
but for code homogeneities, I think you can reuse the `retrier` function that we already use: "github.com/cenkalti/backoff"

here is an example that use it https://github.com/DataDog/datadog-agent/blob/92d84a20a482fe56626a7327b66d9a663671556e/comp/core/tagger/impl-remote/remote.go#L200-L223